### ISSUE TITLE
Webserver to talk to Lightning

### ISSUE THREAD

**Issue by taylordowns2000** [OpenFn/kit#201]
This thing can take workorders (a set of jobs in a workflow and some initial state) from Lightning and process them.

Questions:
1. Does it take a single job at a time?
2. Does it take a whole workorder and process each job sequentially? (Right now that logic is all handled in Lightning and _probably_ should stay there, but there could be significant performance benefits from having it stay in _here_.

Re: Q2, I'm envisioning a middle ground being that the RTM _asks_ Lightning for what to do next after it finishes running a single run.

**Comment by josephjclark** [comment]
The server will have the ability to pull N Attempts  (where n is `MAX_CONCURRENT_ATTEMPS - CURRENT_ACTIVE_ATTEMPTS`) at a time.

It takes the whole work order in one go and runs each job sequentially. The ability to traverse a workflow graph actually needs to be pushed down into the Runtime itself.

We will be engineering in capacity to lazy load state and expressions on demand, to reduce the size of the initial load. Lightning will need to prepare an attempt and work out the initial state, the credential for each Job, and indeed decide which Jobs need to run (if retrying from X). This lets us be more performant on the wire, as well as introduce caching in the RTM layer to reduce calls for data it has already seen.

Credentials will always be lazy-loaded from Lightning when a job starts. I don't think the RTM ever needs to save a credential and it can be discarded once the job is done. This does mean that if Lightning is down, a job won't be able to start (but also once a job has started, it'll execute independently of Lightning)

**Comment by josephjclark** [comment]
Closing this down - as it is more or less done in #395. Future work will be spun out to new issues,
